shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,266 @@
+    ///             order to inject its subclass into the accessibility bridge.
+    virtual std::unique_ptr<FlutterPlatformNodeDelegate>
+    CreateFlutterPlatformNodeDelegate() = 0;
+  };
+  //-----------------------------------------------------------------------------
+  /// @brief      Creates a new instance of a accessibility bridge.
+  ///
+  /// @param[in]  user_data           A custom pointer to the data of your
+  ///                                 choice. This pointer can be retrieve later
+  ///                                 through GetUserData().
+  AccessibilityBridge(std::unique_ptr<AccessibilityBridgeDelegate> delegate);
+  ~AccessibilityBridge();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Adds a semantics node update to the pending semantics update.
+  ///             Calling this method alone will NOT update the semantics tree.
+  ///             To flush the pending updates, call the CommitUpdates().
+  ///
+  /// @param[in]  node           A pointer to the semantics node update.
+  void AddFlutterSemanticsNodeUpdate(const FlutterSemanticsNode* node);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Adds a custom semantics action update to the pending semantics
+  ///             update. Calling this method alone will NOT update the
+  ///             semantics tree. To flush the pending updates, call the
+  ///             CommitUpdates().
+  ///
+  /// @param[in]  action           A pointer to the custom semantics action
+  /// update.
+  void AddFlutterSemanticsCustomActionUpdate(
+      const FlutterSemanticsCustomAction* action);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Flushes the pending updates and applies them to this
+  ///             accessibility bridge. Calling this with no pending updates
+  ///             does nothing, and callers should call this method at the end
+  ///             of an automic batch to avoid leaving the tree in a unstable
+  ///             state. For example if a node reparents from A to B, callers
+  ///             should only call this method when both removal from A and
+  ///             addition to B are in the pending updates.
+  void CommitUpdates();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the flutter platform node delegate with the given id from
+  ///             this accessibility bridge.
+  ///
+  /// @param[in]  id           The id of the flutter accessibility node you want
+  ///                          to retrieve.
+  std::weak_ptr<FlutterPlatformNodeDelegate>
+  GetFlutterPlatformNodeDelegateFromID(ui::AXNode::AXID id) const;
+
+  //------------------------------------------------------------------------------
+  /// @brief      Update the currently focused flutter accessibility node.
```

Comment:
Docs should talk about what kind of focus this is. I'm assuming this is a11y focus, which can be different from e.g. input focus.